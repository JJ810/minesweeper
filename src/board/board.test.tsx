import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { cleanup } from '@testing-library/react';
import Board from './board';

describe('Game Board', () => {
  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it('Should render correctly and match with snapshot', () => {
    const map: string[] = [];
    const tree = renderer.create(<Board map={map} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render welcome message', () => {
    const map: string[] = [];
    const wrapper = mount(<Board map={map} />);
    expect(wrapper.text().includes('Welcome! Press Start button to get started!')).toBe(true);
  });

  it('Should be able to find at least 1 cell', () => {
    const map: string[] = [
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□'
    ];
    const wrapper = mount(<Board map={map} />);
    const element = wrapper.find('cell-1-1');
    expect(element).toBeTruthy();
  });
});
