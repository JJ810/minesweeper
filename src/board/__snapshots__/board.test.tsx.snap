// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Board Should render correctly and match with snapshot 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  Welcome! Press Start button to get started!
</p>
`;
